gobject-introspection: Fix macos shared lib paths

hamishmack opened the pull request:
Fixes #40599.  This is the similar to #46310, but updated for the gnome-3.30 branch.

I used `cat $(nix-build ./default.nix -A atk.dev --system x86_64-darwin)/share/gir-1.0/Atk-1.0.gir | grep shared-library` to confirm the issue was still present in the gnome-3.30 branch and to confirm that this patch fixes it.

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

jtojnar committed:
gnome-recipes: init at 2.0.2

jtojnar committed:
libhandy: init at 0.0.4

jtojnar committed:
meson: 0.46.1 → 0.48.0

jtojnar committed:
glib-networking: add python3 for install script

Meson no longer propagates it so we need to re-add it.

jtojnar committed:
gnome3.gnome-settings-daemon: add python3 for install script

Meson no longer propagates it so we need to re-add it.

jtojnar committed:
gnome3.gnome-bluetooth: add python3 for install script

Meson no longer propagates it so we need to re-add it.

jtojnar committed:
gnome3.gnome-session: add python3 for install script

Meson no longer propagates it so we need to re-add it.

jtojnar committed:
gst_all_1.gst-plugins-base: add python3 for install script

Meson no longer propagates it so we need to re-add it.

jtojnar committed:
gst_all_1.gst-plugins-bad: add python3 for install script

Meson no longer propagates it so we need to re-add it.

jtojnar committed:
libinput: add python3 for install script

Meson no longer propagates it so we need to re-add it.

jtojnar committed:
vala_0_42: init at 0.42.2

jtojnar committed:
webkitgtk: 2.20.5 → 2.22.0

jtojnar committed:
telepathy-glib: clean up

jtojnar committed:
vala: enable parallel building

jtojnar committed:
libpsl: 0.17.0 → 0.20.2

jtojnar committed:
gnome3.adwaita-icon-theme: 3.28.0 → 3.30.0

jtojnar committed:
gnome3.aisleriot: 3.22.5 → 3.22.6

jtojnar committed:
atk: 2.28.1 → 2.30.0

jtojnar committed:
baobab: 3.28.0 → 3.30.0

jtojnar committed:
gnome3.gnome-notes: 3.28.3 → 3.30.2

jtojnar committed:
gnome3.cheese: 3.28.0 → 3.30.0

jtojnar committed:
gnome3.dconf: 0.28.0 → 0.30.0

jtojnar committed:
gnome3.dconf-editor: 3.28.0 → 3.30.2

jtojnar committed:
epiphany: 3.28.3.1 → 3.30.1

jtojnar committed:
evince: 3.28.2 → 3.30.1

jtojnar committed:
gnome3.evolution: 3.28.5 → 3.30.1

jtojnar committed:
evolution-data-server: 3.28.5 → 3.30.1

jtojnar committed:
gnome3.file-roller: 3.28.1 → 3.30.1

jtojnar committed:
gnome3.five-or-more: 3.28.0 → 3.30.0

jtojnar committed:
gdk_pixbuf: 2.36.12 → 2.38.0

jtojnar committed:
gnome3.gdm: 3.28.3 → 3.30.1

jtojnar committed:
gnome3.gedit: 3.28.1 → 3.30.2

jtojnar committed:
gnome3.gjs: 1.52.3 → 1.54.1

jtojnar committed:
glib: 2.56.0 → 2.58.1

jtojnar committed:
glib-networking: 2.56.0 → 2.58.0

jtojnar committed:
gnome3.gnome-backgrounds: 3.28.0 → 3.30.0

jtojnar committed:
gnome3.gnome-boxes: 3.28.5 → 3.30.1

jtojnar committed:
gnome3.gnome-calculator: 3.28.2 → 3.30.1

jtojnar committed:
gnome3.gnome-calendar: 3.28.2 → 3.30.0

jtojnar committed:
gnome3.gnome-characters: 3.28.2 → 3.30.0

jtojnar committed:
gnome3.gnome-chess: 3.28.1 → 3.30.0

jtojnar committed:
gnome3.gnome-clocks: 3.28.0 → 3.30.1

jtojnar committed:
gnome3.gnome-color-manager: 3.28.0 → 3.30.0

jtojnar committed:
gnome3.gnome-contacts: 3.28.2 → 3.30.1

jtojnar committed:
gnome3.gnome-control-center: 3.28.2 → 3.30.1

jtojnar committed:
gnome3.gnome-desktop: 3.28.2 → 3.30.1

jtojnar committed:
gnome3.gnome-disk-utility: 3.28.3 → 3.30.1

jtojnar committed:
gnome3.gnome-documents: 3.28.2 → 3.30.0

jtojnar committed:
gnome3.gnome-logs: 3.28.5 → 3.30.0

jtojnar committed:
gnome3.gnome-maps: 3.28.2 → 3.30.1

jtojnar committed:
gnome3.gnome-mines: 3.28.0 → 3.30.1.1

jtojnar committed:
gnome3.gnome-music: 3.28.2.1 → 3.30.1

jtojnar committed:
gnome3.gnome-online-accounts: 3.28.0 → 3.30.0

jtojnar committed:
gnome3.gnome-online-miners: 3.26.0 → 3.30.0

jtojnar committed:
gnome3.gnome-packagekit: 3.28.0 → 3.30.0

jtojnar committed:
gnome3.gnome-photos: 3.28.0 → 3.30.1

jtojnar committed:
gnome3.gnome-power-manager: 3.26.0 → 3.30.0

jtojnar committed:
gnome3.gnome-screenshot: 3.26.0 → 3.30.0

jtojnar committed:
gnome3.gnome-session: 3.28.1 → 3.30.1

jtojnar committed:
gnome3.gnome-settings-daemon: 3.28.1 → 3.30.1.2

jtojnar committed:
gnome3.gnome-shell: 3.28.3 → 3.30.1

jtojnar committed:
gnome3.gnome-shell-extensions: 3.28.1 → 3.30.1

jtojnar committed:
gnome3.gnome-software: 3.28.2 → 3.30.3

jtojnar committed:
gnome3.gnome-sudoku: 3.28.0 → 3.30.0

jtojnar committed:
gnome3.gnome-system-monitor: 3.28.2 → 3.30.0

jtojnar committed:
gnome3.gnome-taquin: 3.28.0 → 3.30.0

jtojnar committed:
gnome3.gnome-terminal: 3.28.2 → 3.30.1

jtojnar committed:
gnome3.gnome-tweaks: 3.28.1 → 3.30.1

jtojnar committed:
gobjectIntrospection: 1.56.0 → 1.58.0

jtojnar committed:
gsettings-desktop-schemas: 3.28.0 → 3.28.1

jtojnar committed:
gnome3.gtksourceview: 3.24.6 → 3.24.8

jtojnar committed:
gucharmap: 11.0.1 → 11.0.2

jtojnar committed:
gvfs: 1.36.2 → 1.38.1

jtojnar committed:
gnome3.iagno: 3.28.0 → 3.30.0

jtojnar committed:
libdazzle: 3.28.5 → 3.30.1

jtojnar committed:
librsvg: 2.42.4 → 2.44.7

jtojnar committed:
libsecret: 0.18.5 → 0.18.6

jtojnar committed:
libsoup: 2.62.2 → 2.64.1

jtojnar committed:
gnome3.lightsoff: 3.28.0 → 3.30.0

jtojnar committed:
gnome3.metacity: 3.28.0 → 3.30.0

jtojnar committed:
gnome3.mutter: 3.28.3 → 3.30.1

jtojnar committed:
gnome3.nautilus: 3.28.1 → 3.30.2

jtojnar committed:
orca: 3.28.2 → 3.30.1

jtojnar committed:
gnome3.polari: 3.28.1 → 3.30.1

jtojnar committed:
gnome3.seahorse: 3.20.0 → 3.30

jtojnar committed:
shotwell: 0.28.2 → 0.30.0

jtojnar committed:
simple-scan: 3.28.1 → 3.30.1.1

jtojnar committed:
gnome3.sushi: 3.28.3 → 3.30.0

jtojnar committed:
gnome3.swell-foop: 3.28.0 → 3.30.0

jtojnar committed:
gnome3.tracker-miners: 2.1.3 → 2.1.5

jtojnar committed:
gnome3.vte: 0.52.2 → 0.54.1

jtojnar committed:
yelp: 3.28.1 → 3.30.0

jtojnar committed:
gnome3.yelp-xsl: 3.28.0 → 3.30.1

jtojnar committed:
gtk3: 3.22.30 → 3.24.1

jtojnar committed:
gnome3.gconf: remove

The throw statement broke building gnome3 attribute set. Since 18.09
has already been branched off, we just remove the warning.

jtojnar committed:
libhttpseverywhere: fix build

jtojnar committed:
python2.pkgs.gst-python: fix build

hedning committed:
spidermonkey_60: init at 60.2.0

A new test checks and fails if we're building in js/src, so we create a build
dir like Arch does.

hedning committed:
sysprof: 3.28.1 → 3.30.0

jtojnar committed:
gnome3.atomix: 3.29.3 → 3.30.0.1

jtojnar committed:
gnome3.gnome-getting-started-docs: 3.28.2 → 3.30.0

jtojnar committed:
gnome3.gnome-user-docs: 3.28.2 → 3.30.1

hedning committed:
nixos/gdm: use XDG_DATA_DIRS to find sessions

Gdm now searches for session files in XDG_DATA_DIRS so we no longer need the
sessions_dir.patch.

jtojnar committed:
pitivi: 0.99.0 → 0.999.0

hedning committed:
gobjectIntrospection: Never strip away absolute paths

Upstream now strips absolute paths to their basename on all platforms apart from
Darwin: https://gitlab.gnome.org/GNOME/gobject-introspection/commit/a41abe1868a693387cd5cf85567cf2e0fd6c62df

To get around this without modifying the basename test we simply pass in
`basename=False` when normally generating gir files.

LnL7 commented:
@GrahamcOfBorg build gobject-introspection

GrahamcOfBorg commented:
<!--REQUEST_ID=12cc111b-5e19-4e19-9f40-d26f54413baf-->
No attempt on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46312&attempt_id=f12ba2a5-4a1d-4fa9-bbbf-26743cb582f3)

The following builds were skipped because they don't evaluate on x86_64-linux: gobject-introspection

<details><summary>Partial log (click to expand)</summary><p>

```
Cannot nix-instantiate `gobject-introspection' because:
error: attribute 'gobject-introspection' in selection path 'gobject-introspection' not found

```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=12cc111b-5e19-4e19-9f40-d26f54413baf-->
No attempt on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46312&attempt_id=e74b3bb0-c79e-4af6-8506-ecd69d45123b)

The following builds were skipped because they don't evaluate on aarch64-linux: gobject-introspection

<details><summary>Partial log (click to expand)</summary><p>

```
Cannot nix-instantiate `gobject-introspection' because:
error: attribute 'gobject-introspection' in selection path 'gobject-introspection' not found

```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=12cc111b-5e19-4e19-9f40-d26f54413baf-->
No attempt on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46312&attempt_id=5ba0c5d5-138c-4e8b-a069-e96ebca73a68)

The following builds were skipped because they don't evaluate on x86_64-darwin: gobject-introspection

<details><summary>Partial log (click to expand)</summary><p>

```
Cannot nix-instantiate `gobject-introspection' because:
error: attribute 'gobject-introspection' in selection path 'gobject-introspection' not found

```
</p></details>

hedning committed:
pythonPackages.pygobject3: 3.26.1 -> 3.30.1

hedning committed:
gnome3.gnome-panel: 3.28.0 → 3.30.0

hedning committed:
gnome3.geary: fix build with webkit-2.22.0

hedning committed:
gnome3.gnome-applets: 3.28.0 → 3.30.0

hedning committed:
gnome3.gnome-flashback: 3.28.0 → 3.30.0

hedning committed:
gnome3.gnome-usage: 3.28.0 → 3.30.0

hedning committed:
libunique: Don't fail on deprecated-declarations

jtojnar committed:
gnome3.folks: fix build

jtojnar committed:
gnome3.gpaste: 3.28.2 → 3.30.1

jtojnar committed:
zeitgeist: fix build

jtojnar committed:
gnome3.gnome-devel-docs: 3.28.0 → 3.30.1

jtojnar committed:
gnome3.pomodoro: fix build

jtojnar committed:
gnome-builder: 3.28.4 → 3.30.0

jtojnar committed:
jsonrpc-glib: 3.28.1 → 3.30.0

jtojnar committed:
at-spi2-atk: 2.26.2 → 2.30.0

jtojnar committed:
at-spi2-core: 2.28.0 → 2.30.0

jtojnar committed:
gnomeExtensions.dash-to-dock: 63 → 64

jtojnar committed:
gnomeExtensions.dash-to-panel: 11 → 16

jtojnar committed:
farstream: remove gst-python dependency

hedning committed:
gnomeExtensions.icon-hider: 19 -> 20

hedning committed:
gnomeExtensions.nohotcorner: 16.0 -> 17.0

hedning committed:
gnomeExtensions.system-monitor: 33 -> 36

jtojnar committed:
gnome-latex: 3.30.1 → 3.30.2

jtojnar committed:
gtksourceview4: 4.0.0 → 4.0.3

jtojnar committed:
empathy: fix build

hedning committed:
glib: use absolute path for `gio-launch-desktop`

glib tries to look up `gio-launch-desktop` in the path which obviously can fail.
Specifically this now happens when launching gnome-shell.

hamishmack commented:
I have rebased this on latest gnome-3.30 branch.

jtojnar committed:
template-glib: 3.28.0 → 3.30.0

jtojnar committed:
python3.pkgs.pyatspi: 2.26.0 → 2.30.0

hedning committed:
playerctl: switch to meson and fix build

worldofpeace committed:
gnome3.mutter328: init

hedning committed:
xfce.xfce4-namebar-plugin: Specify vala_0_40

hedning committed:
telepathy-salut: use standard pkgconfig

Building glib with selinux support caused the build to fail due to selinux being
specificed in `Required.private` in gio-2.0.pc. `pkgconfigUpstream` isn't
patched to handle `Required.private` properly.

hedning committed:
planner: don't error on deprecrated declarations

hedning committed:
spice-up: 1.3.2 -> 1.6.0

Doesn't build with default vala, so we specify 0.40.

hedning committed:
ldmtool: don't fail on deprecated declarations

hedning committed:
gnucash24: add `pcre` fix build

hedning committed:
gnucash: don't fail on glib-2.58's new deprecrations

hedning committed:
purple-facebook: Remove unused mercurial dependency

Having python3 in the PATH and python2's setuptools and enum in PYTHONPATH
breaks any python3 script. Having mercurial in buildInputs makes this
condition true, breaking glib's python scripts, which after 2.58 use python3.

Mercurial isn't actually needed in buildInputs, so removing mercurial is a
simple fix.

jtojnar committed:
gnome-podcasts: init at 0.4.5

jtojnar committed:
gnome3.grilo-plugins: 0.3.7 → 0.3.8

jtojnar committed:
systemd: fix build with meson 0.48

jonafato committed:
gnomeExtensions.icon-hider: 20 -> 21

jonafato committed:
gnomeExtensions.nohotcorner: 17.0 -> 18.0

jtojnar committed:
libratbag: fix meson/python compatibility

jonafato committed:
gnomeExtensions.caffeine: unstable-2017-06-21 -> unstable-2018-09-25

jtojnar committed:
gnome3.eog: 3.28.3 → 3.28.4

jtojnar committed:
json-glib: 1.4.2 → 1.4.4

jtojnar committed:
gnome3.libgda: 5.2.4 → 5.2.5

jtojnar committed:
gnome3.tracker: 2.1.4 → 2.1.5

jtojnar committed:
gnome3.gnome-remote-desktop: init at 0.1.6

jtojnar committed:
nixos/gnome3: enable remote desktop on wayland

samueldr commented:
There is, again, a need for a rebase.

@jtojnar this branch is pretty much yours; any opinions on the fix?

jtojnar commented on pkgs/development/libraries/gobject-introspection/default.nix:
```diff
@@ -40,6 +40,7 @@ stdenv.mkDerivation rec {
   setupHook = ./setup-hook.sh;
 
   patches = [
+    ./macos-shared-library.patch
```
Could you name the patch something like `macos-absolutize-relocatable-install-name` and add a coment with a brief description here?

hamishmack committed:
gobject-introspection: Fix macos shared lib paths

Uses patch Nirbheek Chauhan's patch from this issue upstream issue:
https://gitlab.gnome.org/GNOME/gobject-introspection/issues/222

Fixes #40599

hamishmack commented:
Updated as suggested and rebased

hedning committed:
gxneur: don't fail on new glib deprecration

hedning committed:
hardinfo: fix build with new libsoup

hedning committed:
gitg: 3.26.0 -> 3.30.0

hedning committed:
gnome3.zenity: 3.28.1 -> 3.30.0

jonafato committed:
gnomeExtensions.no-title-bar: 8 -> 9

jtojnar commented:
https://github.com/NixOS/nixpkgs/pull/46310 was merged already.
